compiler/frontend-spark/src/main/java/edu/snu/onyx/compiler/frontend/spark/JavaRDD.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.onyx.compiler.frontend.spark;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import edu.snu.onyx.client.JobLauncher;
+import edu.snu.onyx.common.dag.DAG;
+import edu.snu.onyx.common.dag.DAGBuilder;
+import edu.snu.onyx.common.ir.edge.IREdge;
+import edu.snu.onyx.common.ir.edge.executionproperty.DataCommunicationPatternProperty;
+import edu.snu.onyx.common.ir.edge.executionproperty.KeyExtractorProperty;
+import edu.snu.onyx.common.ir.vertex.IRVertex;
+import edu.snu.onyx.common.ir.vertex.InitializedSourceVertex;
+import edu.snu.onyx.common.ir.vertex.LoopVertex;
+import edu.snu.onyx.common.ir.vertex.OperatorVertex;
+import edu.snu.onyx.common.ir.vertex.executionproperty.ParallelismProperty;
+import edu.snu.onyx.compiler.frontend.spark.coder.SparkCoder;
+import edu.snu.onyx.compiler.frontend.spark.transform.MapTransform;
+import edu.snu.onyx.compiler.frontend.spark.transform.ReduceTransform;
+import edu.snu.onyx.compiler.frontend.spark.transform.SerializableBinaryOperator;
+import edu.snu.onyx.compiler.frontend.spark.transform.SerializableFunction;
+import org.apache.spark.serializer.KryoSerializer;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Stack;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Java RDD.
+ * @param <T> type of the final element.
+ */
+public final class JavaRDD<T extends Serializable> {
+  private final SparkContext sparkContext;
+  private final Integer parallelism;
+  private final Stack<LoopVertex> loopVertexStack;
+  private final DAG<IRVertex, IREdge> dag;
+  @Nullable private final IRVertex lastVertex;
```

Comment:
Please add a TODO for supporting arbitrary RDD DAGs.
I think we are translating a list of RDDs at the moment.